tests/functional_tests/cpp_backend_tests/fvm_nabla.py

```diff
@@ -0,0 +1,78 @@
+import sys
+
+from functional.iterator.backends.gtfn.backend import generate
+from functional.iterator.builtins import *
+from functional.iterator.runtime import closure, fundef, offset
+from functional.iterator.tracing import trace
+
+
+E2V = offset("E2V")
+V2E = offset("V2E")
+
+
+@fundef
+def compute_zavgS(pp, S_M):
+    zavg = 0.5 * (deref(shift(E2V, 0)(pp)) + deref(shift(E2V, 1)(pp)))
+    return make_tuple(tuple_get(0, deref(S_M)) * zavg, tuple_get(1, deref(S_M)) * zavg)
+
+
+def _unroll_reduce(binop, init, n):
```

Comment:
Should be an IR-pass I guess.

Reply:
I don't remember if we concluded that iterator ir doesn't have reduce or if we decided to keep it. Let's have a chat later.